internal/mapdecode/decode.go

```diff
@@ -129,9 +129,17 @@ func decodeFrom(opts *options, src interface{}) Into {
 		cfg := mapstructure.DecoderConfig{
 			ErrorUnused: !opts.IgnoreUnused,
 			Result:      dest,
-			DecodeHook: mapstructure.ComposeDecodeHookFunc(
-				mapstructure.StringToTimeDurationHookFunc(),
-				decoderDecodeHook(opts),
+			DecodeHook: fromReflectHook(
+				supportPointers(
+					composeReflectHooks(
+						unmarshalerHook(opts),
+
+						// durationHook must come before the strconvHook
+						// because the Kind of time.Duration is Int64.
+						durationHook,
```

Comment:
You'll get [zap.LevelEncoder](https://github.com/uber-go/zap/blob/master/zapcore/encoder.go#L68) for free :)